Dockerfile.commonv8.2

```diff
@@ -0,0 +1,335 @@
+
+### ROS
+# download/build ROS from source
+RUN python3 -m pip install distro
+RUN mkdir -p ${ROS_ROOT}/src && \
+    cd ${ROS_ROOT} && \
+    # https://answers.ros.org/question/325245/minimal-ros2-installation/?answer=325249#post-id-325249
+    rosinstall_generator --deps --rosdistro ${ROS_DISTRO} ${ROS_PKG} \
+		launch_xml \
+		launch_yaml \
+		launch_testing \
+		launch_testing_ament_cmake \
+		demo_nodes_cpp \
+		demo_nodes_py \
+		example_interfaces \
+		camera_calibration_parsers \
+		camera_info_manager \
+		cv_bridge \
+		v4l2_camera \
+		vision_opencv \
+		vision_msgs \
+		image_geometry \
+		image_pipeline \
+		image_transport \
+		compressed_image_transport \
+		compressed_depth_image_transport \
+		> ros2.${ROS_DISTRO}.${ROS_PKG}.rosinstall && \
+    cat ros2.${ROS_DISTRO}.${ROS_PKG}.rosinstall && \
+    vcs import src < ros2.${ROS_DISTRO}.${ROS_PKG}.rosinstall && \
+    # install dependencies using rosdep
+    apt-get update && \
+    cd ${ROS_ROOT} && \
+    rosdep init && \
+    rosdep update && \
+    rosdep install -y \
+    	  --ignore-src \
+       --from-paths src \
+	  --rosdistro ${ROS_DISTRO} \
+	  --skip-keys "libopencv-dev libopencv-contrib-dev libopencv-imgproc-dev python-opencv python3-opencv" && \
+    rm -rf /var/lib/apt/lists/* && \
+    apt-get clean && \
+    # build it!
+    colcon build \
+        --merge-install \
+        --cmake-args -DCMAKE_BUILD_TYPE=Release && \
+    # remove build files
+    rm -rf ${ROS_ROOT}/src && \
+    rm -rf ${ROS_ROOT}/logs && \
+    rm -rf ${ROS_ROOT}/build && \
+    rm ${ROS_ROOT}/*.rosinstall
+    
+#
+# Set the default DDS middleware to cyclonedds
+# https://github.com/ros2/rclcpp/issues/1335
+#
+ENV RMW_IMPLEMENTATION=rmw_cyclonedds_cpp
+
+### boooost
+RUN wget https://downloads.sourceforge.net/project/boost/boost/1.78.0/boost_1_78_0.tar.gz \
+    && tar xfz boost_1_78_0.tar.gz \
+    && rm boost_1_78_0.tar.gz \
+    && cd boost_1_78_0 \
+    && ./bootstrap.sh --prefix=/usr/local --with-libraries=program_options \
+    && ./b2 install \
+    && cd .. \
+    && rm -rf boost_1_78_0
+
+### build joystick drivers
+RUN git clone -b foxy-devel https://github.com/ros-drivers/joystick_drivers && \
+    cd joystick_drivers && \
+    source /opt/ros/$ROS_DISTRO/install/setup.bash && \
+    colcon build --merge-install --install-base /opt/ros/foxy/install --packages-up-to joy --cmake-args -DCMAKE_BUILD_TYPE=Release && \
+    cd .. && rm -rf joystick_drivers
+    
+# 
+# setup entrypoint
+#
+COPY ./ros_entrypoint.sh /ros_entrypoint.sh
+
+RUN sed -i \
+    's/ros_env_setup="\/opt\/ros\/$ROS_DISTRO\/setup.bash"/ros_env_setup="${ROS_ROOT}\/install\/setup.bash"/g' \
+    /ros_entrypoint.sh && \
+    cat /ros_entrypoint.sh
+
+RUN echo 'source ${ROS_ROOT}/install/setup.bash' >> /root/.bashrc
+
+
+# clean up because i forgot to do it earlier and dont want to rebuild the entire image :(
+RUN rm -rf \
```

Comment:
I mean yeah but the files you delete here stay in the binaries because layer history